project-base/app/src/Model/Product/Product.php

```diff
@@ -43,6 +43,7 @@
 class Product extends BaseProduct
 {
     public const PDF_SUFFIX = '.pdf';
+    public const RELATED_PRODUCTS_FRONTEND_LIMIT = 30;
```

Comment:
The Frontend limits are scattered across the codebase quite randomly (Facade, Query, Entity, ...). 
It doesn't feel right for me. 

I would recommend putting every frontend limit constant to the same layer of the application.